diff --git a/app/models/time_entry_query.rb b/app/models/time_entry_query.rb
index a612ba7..dea4cdf 100644
--- a/app/models/time_entry_query.rb
+++ b/app/models/time_entry_query.rb
@@ -33,6 +33,7 @@ class TimeEntryQuery < Query
     QueryAssociationColumn.new(:issue, :category, :caption => :field_category, :sortable => "#{IssueCategory.table_name}.name"),
     QueryColumn.new(:comments),
     QueryColumn.new(:hours, :sortable => "#{TimeEntry.table_name}.hours", :totalable => true),
+    QueryColumn.new(:updated_on, :sortable => "#{TimeEntry.table_name}.updated_on", :default_order => 'desc'),
   ]
 
   def initialize(attributes=nil, *args)
@@ -42,6 +43,8 @@ class TimeEntryQuery < Query
 
   def initialize_available_filters
     add_available_filter "spent_on", :type => :date_past
+    add_available_filter "updated_on", :type => :date_past
+    add_available_filter "created_on", :type => :date_past
 
     add_available_filter("project_id",
       :type => :list, :values => lambda { project_values }
